lambdex/resources/lambdex_handler.py

```diff
@@ -25,10 +25,13 @@
 try:
     # PEX >= 1.6.0
     from pex.pex_bootstrapper import bootstrap_pex_env
-    from pex.third_party.pkg_resources import EntryPoint as __EntryPoint
 except ImportError:
     # PEX < 1.6.0 has an install requirement of setuptools which we leverage knowledge of.
     from _pex.pex_bootstrapper import bootstrap_pex_env
+
```

Comment:
Is this change necessary? The only behavior difference I can see between the two cases would be if it were possible to get `bootstrap_pex_env` from PEX but not `EntryPoint` . And AFAICT that should not be possible - they should both ride together. If you revert this part do things not work?

Reply:
Right now this is what's causing me grief 

```python
try:
    # PEX >= 1.6.0
    from pex.pex_bootstrapper import bootstrap_pex_env
    from pex.third_party.pkg_resources import EntryPoint as __EntryPoint
except ImportError:
    # PEX < 1.6.0 has an install requirement of setuptools which we leverage knowledge of.
    from _pex.pex_bootstrapper import bootstrap_pex_env
    from pkg_resources import EntryPoint as __EntryPoint
```

```python
>>> from pex.pex_bootstrapper import bootstrap_pex_env
```

Succeeds, but the second fails

```python
>>> from pex.third_party.pkg_resources import EntryPoint as __EntryPoint
ImportError: cannot import name 'six' from 'pex.vendor._vendored.setuptools.pkg_resources.extern' (/home/jackevans/code/lambdex/.venv/lib/python3.12/site-packages/pex/vendor/_vendored/setuptools/pkg_resources/extern/__init__.py)
```

In this sitaution I don't think we want to fall back to running 

```
>>> from _pex.pex_bootstrapper import bootstrap_pex_env
```

Because this gives a different `ImportError`

```
>>> from _pex.pex_bootstrapper import bootstrap_pex_env
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named '_pex'
```

So that's why I've had to split it into x2 try/catch statements.

Reply:
Ah OK, so the case that I assumed can't happen is exactly what is happening. Shows what I know...

Reply:
Do you know which version of PEX is in play here?

Reply:
Ah, `pex-2.3.1`

Reply:
Well, if this works for you then it's probably fine. 

But for the record - this was all thin-ice stuff to begin with. The only stable API PEX guarantees is its command-line interface. The internals can change at any time. So importing from PEX was always shady. Not your responsibility though!